packages/core/src/rpc/utils.ts

```diff
@@ -22,6 +22,8 @@ export const getMessageKey = (hash: Buffer | string, message: Message) => {
 		key.writeUintBE(message.payload.timestamp * 2 + 1, 0, 6)
 	} else if (message.type === "session") {
 		key.writeUintBE(message.payload.sessionIssued * 2, 0, 6)
+	} else if (message.type === "customAction") {
+		throw Error("getMessageKey is not yet implemented for customAction!")
```

Comment:
Making a note here as a TODO - we probably want to hash custom actions to get a message key

Reply:
does the existing hashing approach (for session and action) assume that we never have messages that have the same timestamp? what happens if there is a collision?

Reply:
ok, the message key also includes the hash of the action, we can just set the timestamp to zero